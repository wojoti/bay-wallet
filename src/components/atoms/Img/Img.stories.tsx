import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Img, ImgType} from 'components/atoms';

export default {
  title: 'Atoms/Img',
  component: Img,
} as ComponentMeta<typeof Img>;
const Template: ComponentStory<typeof Img> = args => <Img {...args} />;

export const CardWithLock = Template.bind({});
CardWithLock.args = {
  src: ImgType.cardLock,
  width: 100,
  height: 100,
  alt: 'cardLock',
  testId: '',
};
