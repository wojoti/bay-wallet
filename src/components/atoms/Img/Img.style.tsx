import {StyleSheet} from 'react-native';

interface Props {
  width: number;
  height: number;
}
const style = ({width, height}: Props) =>
  StyleSheet.create({
    wrapper: {
      margin: 0,
      padding: 0,
      width: width,
      height: height,
    },
  });
export default style;
