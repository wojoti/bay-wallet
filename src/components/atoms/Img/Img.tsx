import Card_lock from 'img/card_lock.svg';
import {PropsWithChildren} from 'react';

export type ImgProps = PropsWithChildren<{
  src: ImgType;
  width: number;
  height: number;
  alt: string;
  testId?: string;
}>;

export enum ImgType {
  cardLock = 'img/card_lock.svg',
  //  logo = 'img/bw_logo.Img',
}

function Img({src, width, height, testId}: ImgProps) {
  switch (src) {
    case ImgType.cardLock:
      return <Card_lock width={width} height={height} testID={testId} />;
    // case ImgType.logo:
    //   return <Logo width={width} height={height} testID={testId} />;
  }
}
export default Img;
