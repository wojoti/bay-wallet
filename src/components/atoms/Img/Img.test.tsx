import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {Img, ImgType} from '..';
import {ImgProps} from './Img';

const props: ImgProps = {
  src: ImgType.cardLock,
  width: 100,
  height: 100,
  alt: 'alt-text',
  testId: 'test-img-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Img {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render Img', () => {
  render(<Img {...props} />);
  const ImgElement = screen.getByTestId('test-img-id');
  expect(ImgElement).toBeOnTheScreen();
});
