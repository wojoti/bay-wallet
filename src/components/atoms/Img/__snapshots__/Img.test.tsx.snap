// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<SvgMock
  height={100}
  testID="test-img-id"
  width={100}
/>
`;
